acceptance-tests/README.md

```diff
@@ -16,37 +16,71 @@ normally of the form `test_[timestamp]`. This makes it easy to identify the test
 name collisions. If the test runs successfully to completion, the project is deleted at the end of the test.
 If the test fails, then the project is left around to aid debugging.
 
+## Running on Docker
 
-## Running over Docker
+The easiest way to run the tests is to use docker/docker-compose.
 
-The easiest way to run the tests is to use docker/docker-compose. The docker-compose environment is made up of
-two containers: one hosting selenium, and one running the tests.
+### Installation
 
+The `docker-compose` command has to be available: it is easiest to install it with `pip`.
 
-#### Installation
-The `docker-compose` command has to be available. For installation steps see: https://docs.docker.com/compose/install/.
+### Running
 
-#### Running
-For running with docker, the parameters to the test need to be provided as environment variables:
+To run through docker-compose, the tests parameters need to be provided as environment variables:
 
 ```
-docker-compose run -e RENKU_TEST_URL=https://renku-kuba.dev.renku.ch -e RENKU_TEST_FULL_NAME="<full user name>" -e RENKU_TEST_EMAIL=<email> -e RENKU_TEST_USERNAME=<username> -e RENKU_TEST_PASSWORD=<password> sbt
+docker-compose run \
+-e RENKU_TEST_URL=https://renku-kuba.dev.renku.ch \
+-e RENKU_TEST_FULL_NAME="<full user name>" \
+-e RENKU_TEST_EMAIL=<email> \
+-e RENKU_TEST_USERNAME=<username> \
+-e RENKU_TEST_PASSWORD=<password> \
+sbt
 ```
 
 By default, all tests are run. You can provide an argument to run a specific test:
-
 ```
-docker-compose run -e RENKU_TEST_URL=https://renku-kuba.dev.renku.ch -e RENKU_TEST_FULL_NAME="<full user name>"
--e RENKU_TEST_EMAIL=<email> -e RENKU_TEST_USERNAME=<username> -e RENKU_TEST_PASSWORD=<password> sbt
-/tests/docker-run-tests.sh *<test-class-name>*
+docker-compose run \
+-e RENKU_TEST_URL=https://renku-kuba.dev.renku.ch \
+-e RENKU_TEST_FULL_NAME="<full user name>" \
+-e RENKU_TEST_EMAIL=<email> \
+-e RENKU_TEST_USERNAME=<username> \
+-e RENKU_TEST_PASSWORD=<password> \
+sbt *<test-class-name>*
 ```
 __**IMPORTANT**__
 
 Once you do changes to the `Dockerfile` do not forget to build the image with `docker-compose build`.
 
+## Running through Helm test
+
+The `helm test` command can also be used to execute the tests against a Helm release of Renku.
+
+Suppose to have a deployed release of Renku called `renku-foo`, the `values.yml` file should
+contain the credential to an user that needs to be already present on `dev.renku.ch`
```

Comment:
Must that user already be present in keycloak of `renku-foo` too or is it automatically created there? In general I think it's good practice to assume that any person who is not an SDSC dev should be able to read and follow this documentation, so I'd not assume that everybody reading this docs runs their tests against an instance that uses `dev.renku.ch/gitlab` as GitLab backend. We could write something like "When testing a Renku release that relies on an external GitLab instance, the test user must already have an existing account on that GitLab instance with login credentials specified in the values.yaml file."

My preferred end state would be:
- For regular deployments that do not depend on another gitlab instance, the test user is created in the beginning through the keycloak API and (if tests are successful) destroyed at the very end together with all the projects that were created during tests.
- For a deployment which does rely on another gitlab instance, the GitLab credentials of and existing user for that instance must be provided through the values.yaml file.

This would allow us to get rid of the same user with the same password having to be present in two separate locations.

Reply:
I have updated the README as you suggested.
I'll tackle the all story about user generation in a following PR (perhaps also after some team's discussion).